src/wallet/wallet.cpp

```diff
@@ -3929,7 +3929,7 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
 
     int64_t nStart = GetTimeMillis();
     bool fFirstRun = true;
-    CWallet *walletInstance = new CWallet(name, CWalletDBWrapper::Create(path));
+    std::unique_ptr<CWallet> walletInstance(new CWallet(name, CWalletDBWrapper::Create(path)));
```

Comment:
Use MakeUnique?

Reply:
Why? This is exactly what MakeUnique does, except that this constructs one unique_ptr, whereas `std::unique_ptr<CWallet> walletInstance = MakeUnique<...>(...)` would construct two.

Reply:
Why do you say it would construct two?

Reply:
`MakeUnique` constructs one (the right-hand side), then the left hand side is created using the move constructor.

Reply:
I think the compiler will optimize the copy initialization. Anyway, there are [several advantages](https://stackoverflow.com/a/37514601) and we are using MakeUnique so that we can replace with std::make_unique when we switch to c++14.

Reply:
Ah, you're right about the compiler optimization. Didn't realize that. http://en.cppreference.com/w/cpp/language/copy_elision